### DISCUSSION TITLE
How do I uninstall Biome CLI ?

### DISCUSSION THREAD

**Discussion by edumserrano** [biomejs/biome-vscode#271] [Q&A]
I tried out the option in the extension to install Biome CLI:

![image](https://github.com/user-attachments/assets/74647004-0786-49ee-8a62-c125463ed2b6)

But now I want to uninstall it and I don't see any option for that?

**Comment by raphaelgibson** [comment]
I have the sabe problem.

**Comment by nhedger** [comment]
The global binaries are downloaded in [ExtensionContext.storageUri](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext.storageUri)

Practically, this means that they get downloaded into a folder under:

- Linux: `~/.config/Code/User/workspaceStorage`
- macOS: `~/Library/Application\ Support/Code/User/workspaceStorage`
- Windows: `C:\Users\[User]\AppData\Roaming\Code\User\workspaceStorage`

You can find all installed Biome CLIs across all workspaces using `find` (or similar)

```shell
find ~/Library/Application\ Support/Code/User/workspaceStorage -name "biome"
```

It should output something like

```shell
~ find ~/Library/Application\ Support/Code/User/workspaceStorage -name "biome"
/Users/nicolas/Library/Application Support/Code/User/workspaceStorage/35d02d2827d2a00574fa1b3adaed976a/biomejs.biome/biome
/Users/nicolas/Library/Application Support/Code/User/workspaceStorage/3193da92c457eee351225f4b655d66f4/biomejs.biome/biome
```

**Reply by edumserrano** [reply]
This doesn't seem to be enough. I've deleted the folders where `biome.exe` shows up following your instructions but when I start up VSCode in a project that I don't have biome installed as an npm package I still get biome linting errors. I also see VSCode showing that an update to the bundled biome is available:

![image](https://github.com/user-attachments/assets/a6c24715-9759-4ea4-9d27-107271a75d78)

I still can't uninstall biome, this is very annoying. The only thing that works for me at the moment is to uninstall/disable the Biome VSCode extension on projects that I don't want to use Biome...

This question is not closed at all, please open it.

**Reply by nhedger** [reply]
Try looking in `~/Library/Application Support/Code/User/globalStorage/biomejs.biome` too

**Reply by stevan-borus** [reply]
Try to disable Biome for the workspace in which you don't use it

<img width="1208" alt="Screenshot 2024-10-14 at 10 47 27" src="https://github.com/user-attachments/assets/176b1800-8c46-4413-af29-25e017ed7756">

**Reply by edumserrano** [reply]
> Try looking in `~/Library/Application Support/Code/User/globalStorage/biomejs.biome` too

I'm using Windows, I searched for `*biomejs*` and the only hit was at `$env:USERPROFILE\.vscode\extensions\biomejs.biome-2.3.0`. I uninstalled the Biome extension, then deleted that folder. However, after installing the extension and opening VS Code in a project that doesn't have the Biome npm package installed I still get Biome linting and VSCode shows `Biome <version> (bundled)` on the bottom bar.

Still looking for a solution

**Reply by edumserrano** [reply]
> Try to disable Biome for the workspace in which you don't use it
> 
> <img alt="Screenshot 2024-10-14 at 10 47 27" width="1208" src="https://private-user-images.githubusercontent.com/17165251/376168456-176b1800-8c46-4413-af29-25e017ed7756.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MDY1MTAsIm5iZiI6MTcyODkwNjIxMCwicGF0aCI6Ii8xNzE2NTI1MS8zNzYxNjg0NTYtMTc2YjE4MDAtOGM0Ni00NDEzLWFmMjktMjVlMDE3ZWQ3NzU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE0VDExNDMzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0NTU3Mjc5YzgzZjFjZDIzM2E0NGQxNGQ5MGMwMzRkYmY0OWMyYzRhMDIxYzNiNzcxNzljNjY3MzE3NzJiNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3UeV_cXaMX7X38y1RBepSIglrZ8lbRrbjsap6peJWm8">

Surely I don't have to do this for all workspaces I use except the projects were I use Biome right? Surely there should be an uninstall command to uninstall the bundled version no?

**Reply by stevan-borus** [reply]
@edumserrano Yeah.. just hit `Disable (workspace)` in the project where you **do not** use Biome
Same goes for prettier/eslint in other projects where you **do** use Biome

Not sure how to uninstall CLI except to uninstall/disable it via extensions

**Reply by edumserrano** [reply]
> @edumserrano Yeah.. just hit `Disable (workspace)` in the project where you **do not** use Biome Same goes for prettier/eslint in other projects where you **do** use Biome
> 
> Not sure how to uninstall CLI except to uninstall/disable it via extensions

Don't think your example is the same. If you see my original post, I used an option on the biome VSCode extension that allows you to install a bundled Biome version **instead** of installing the Biome NPM package. I was trying things out and that's why I did that. The side effect of doing that is that now for every other project I have, I'm getting Biome linting errors/warnings. I don't want to have to disable the extension for every other project except for the ones where I use Biome.

I want to use Biome only on the projects where I have installed the Biome NPM package. However, now that I've tried out the Biome VSCode option to install a bundled version of Biome I can't seem to do this. This is what I mean by surely there should be an option that does the reverse of whatever the `Select the version of the Biome CLI to install` operation that is performed by the Biome VSCode extension does? Or not?

**Reply by edumserrano** [reply]
To add to the above, whatever the Biome VSCode extension did with the install Biome CLI operation, it only applies to VSCode. Meaning if I type `biome` on my shell it doesn't recognize it. I'm using Windows and it might just be because that operation didn't add it to the `path`  variable or install the package as a global package. 

Anyways, I'm only mentioning this to try and make it clear that all I want to do is reverse that install operation that the Biome VSCode extension did. That's all.

**Reply by stevan-borus** [reply]
I did exactly the same thing.. that is how I got here in the first place. It would definitely be nice if there is an option to uninstall the CLI.
I just wanted to let you know how to mitigate the problem a bit for now.

**Reply by edumserrano** [reply]
Right, thank you @stevan-borus . 

Hopefully someone will provide a resolution for this. @nhedger, you closed this but wouldn't it be best to re-open this until a proper solution is found?

**Reply by aon** [reply]
I found out how to delete the biome binary. When the extension starts it will log the binary path in the output tab, as seen here: https://github.com/biomejs/biome-vscode/blob/d66afc687cdb40728d9861709e6a2f5ff55424dd/src/binary-finder.ts#L247-L251.

From there you can just remove the binary and restart the editor.

**Comment by wsteidley** [comment]
@edumserrano I had the same issue. The root was that I had to do a `find` one level up and found that biome was installed under the `~/.config/Code/User/globalStorage` in addition to `workspaceStorage`. Deleting from both places worked for me
